gm_dotnet_managed/GmodNET/GameConsoleWriter.cs

```diff
@@ -0,0 +1,169 @@
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace GmodNET
+{
+    /// <summary>
+    /// Redirects Console's output to game's console
+    /// </summary>
+    public class GameConsoleWriter : TextWriter
+    {
+        public override string NewLine => "\n";
+        public override Encoding Encoding => Encoding.UTF8;
+
+#pragma warning disable CA2101
+        [DllImport("tier0", CallingConvention = CallingConvention.Cdecl)]
+        private static extern void Msg([MarshalAs(UnmanagedType.LPUTF8Str)] string msg);
+
+        public override void Write(string value)
+        {
+            Msg(new string(value));
+        }
+        public override void Write(char value)
+        {
+            Write(value.ToString());
+        }
+        public override void Write(char[] value)
+        {
+            Write(new string(value));
+        }
+        public override void Write(char[] buffer, int index, int count)
+        {
+            if (buffer == null)
+            {
+                throw new ArgumentNullException(nameof(buffer));
+            }
+            if (index < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(index));
+            }
+            if (count < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(count));
+            }
+            if (buffer.Length - index < count)
+            {
+                throw new ArgumentException();
+            }
+            StringBuilder stringBuilder = new StringBuilder();
+            for (int i = 0; i < count; i++) stringBuilder.Append(buffer[index + i]);
```

Comment:
Use this constructor for string https://docs.microsoft.com/ru-ru/dotnet/api/system.string.-ctor?view=net-5.0#System_String__ctor_System_Char___System_Int32_System_Int32_ instead of all this code.

Reply:
resolved in https://github.com/GmodNET/GmodDotNet/pull/70/commits/415c5f62feca5cd0d2a75590c56c2174f8fe0edd